components/esp_driver_i2c/i2c_master.c

```diff
@@ -808,16 +811,18 @@ static esp_err_t i2c_master_bus_destroy(i2c_master_bus_handle_t bus_handle)
                 }
             }
             bus_handle = NULL;
+            free(i2c_master);
+        } else {
+            free(i2c_master);
```

Comment:
Unless the ESP coding style suggests this, I would refactor the `free(i2c_master);` calls into a single statement at the end end remove two `else` clauses.